common/ManagementTestShared/MgmtRecordTestBase.cs

```diff
@@ -0,0 +1,224 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+
+using Azure.Core;
+using Azure.Core.TestFramework;
+#if RESOURCES_RP
+using Azure.ResourceManager.Resources;
+#else
+using Azure.Management.Resources;
+#endif
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using NUnit.Framework;
+
+
+namespace Azure.ResourceManager.TestFramework
+{
+    [Category("Recorded")]
+    public abstract class MgmtRecordTestBase<TEnvironment> : ClientTestBase where TEnvironment : TestEnvironment, new()
```

Comment:
Can you add xml doc for class, public/protected properties and methods? In class summary, can you explain the difference between existing managementRecordTestBase? Should that one be marked as deprecated?